api/v1beta1/azurecluster_validation.go

```diff
@@ -556,6 +557,23 @@ func validatePrivateDNSZoneName(privateDNSZoneName string, controlPlaneEnabled b
 	return allErrs
 }
 
+// validatePrivateDNSZoneResourceGroup validates the PrivateDNSZoneResourceGroup.
+func validatePrivateDNSZoneResourceGroup(privateDNSZoneName string, privateDNSZoneResourceGroup string, fldPath *field.Path) field.ErrorList {
```

Comment:
This should technically by `validatePrivateDNSZone(...)` since the Private DNS Zone's Resource Group is validated as a regular resource group via `validateResourceGroup(...)`.

I see the rationale behind the naming the function so _"A private DNS Zone's resource group is valid as long as  privateDNSZoneName is provided with the private dns resource group name"_

A non-blocking comment. More of a TODO for us.

Reply:
Understood it. I have added the quotes as a comment for the function.